File: presence/concept-pl-execution-13-fixtures/concept_pulse.cpp
// concept: selective-retention
#include <iostream>
#include <string>
#include <vector>
int main() {
  std::vector<int> signals = {2, 7, 1, 8, 2, 8};
  int threshold = 5;
  int scoreFloor = 100;
  int retainedFloor = 3;
  int score = 0, retained = 0;
  for (std::size_t index = 0; index < signals.size(); ++index) {
    int value = signals[index]; score += value * static_cast<int>(index + 1);
    if (value >= threshold) ++retained;
  }
  std::string verdict = score >= scoreFloor && retained >= retainedFloor ? "coherent" : "fragmented";
  std::cout << verdict << ':' << score << ':' << retained << '\n';
}

File: presence/concept-pl-execution-13-fixtures/concept_pulse.rb
# concept: selective-retention
signals = [2, 7, 1, 8, 2, 8]
threshold = 5
score_floor = 100
retained_floor = 3
score = 0
retained = 0
signals.each_with_index do |value, index|
  score += value * (index + 1)
  retained += 1 if value >= threshold
end
verdict = score >= score_floor && retained >= retained_floor ? "coherent" : "fragmented"
puts "#{verdict}:#{score}:#{retained}"
